static-analysis/autodispose-error-prone-checker/src/main/java/com/uber/autodispose/error/prone/checker/AutoDisposeLeakChecker.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.error.prone.checker;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.ErrorProneFlags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Type;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+import static com.google.errorprone.matchers.Matchers.instanceMethod;
+
+/**
+ * Checker for subscriptions not binding to lifecycle in components with lifecycle.
+ * Use -XepOpt:AutoDisposeLeakCheck flag to add support for custom components with lifecycle.
+ * The sample configuration for Conductor:
+ * <pre><code>
+ *   -XepOpt:AutoDisposeLeakCheck=com.bluelinelabs.conductor.Controller,android.app.Activity
+ * </code></pre>
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    name = "AutoDisposeLeakChecker",
```

Comment:
I left a comment somewhere that I can't find now, but let's call both the bugpattern name and the class `UseAutoDispose`. Don't have `Checker` as a suffix anywhere. This matches the error-prone convention

Reply:
I think maybe that comment wasn't published because I never saw that. Error-prone doesn't have a common stand against it - https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/CompileTimeConstantChecker.java.  Fixing it for this case.

Reply:
https://github.com/uber/AutoDispose/pull/156/commits/282291ba155b03cda9977e5f2bc08e06c51c8e22